src/sqlfluff/core/config.py

```diff
@@ -260,14 +260,24 @@ def load_config_at_path(self, path: str) -> dict:
         self._config_cache[str(path)] = configs
         return configs
 
-    def load_user_appdir_config(self) -> dict:
+    def load_user_appdir_config(self) -> Tuple[Optional[str], dict]:
         """Load the config from the user's OS specific appdir config directory."""
         appname = "sqlfluff"
         appauthor = "sqlfluff"
-        user_config_dir_path = appdirs.user_config_dir(appname, appauthor)
+
+        # On Mac OSX follow Linux XDG base dirs
+        # https://github.com/sqlfluff/sqlfluff/issues/889
+        user_config_dir_path = os.path.expanduser("~/.config/sqlfluff")
+        if appdirs.system == "darwin":
+            appdirs.system = "linux2"
+            user_config_dir_path = appdirs.user_config_dir(appname, appauthor)
+            appdirs.system = "darwin"
+        if not os.path.exists(user_config_dir_path):
+            user_config_dir_path = appdirs.user_config_dir(appname, appauthor)
+
         if os.path.exists(user_config_dir_path):
-            return self.load_config_at_path(user_config_dir_path)
-        return {}
+            return user_config_dir_path, self.load_config_at_path(user_config_dir_path)
```

Comment:
It looks like the first return value exists solely so the automated test can check it. I wonder if we could refactor (split) this into two functions:
* One that returns the directory
* Another that calls the first and returns the config path

I think that may be cleaner, so we're not returning unneeded data to the "real" code. 🤔

Reply:
100%, lazy on my end ^^